Added compiler to README

JosVoigt opened the pull request:
Task 2 of Worksheet 1.

JosVoigt committed:
Added compiler to README

FG-TUM reviewed (approved):
Please don't forget to add build instructions for your code in the README.

JosVoigt committed:
Updated README to better explain usage. Included CMake instructions.

Introduced a basic boost based parser in src/utils/Parser.h
Currently allows for help and 3 settings.

Programmed the planet simualtion by creating a force, position and
velocity calculation according to Störmer-Verlet.

Included doxygen support. Currently always compiles if compiler is
existent (TODO).

Moved header into src/header for less cluttered directories.

Wrote a build/CMakeLists.txt to allow for fluent build.
Currently supports 3/4 options. More info in README.

Added a basic gitignore to prevent build material to enter repository.

JosVoigt committed:
forgot to commit doxygen.cmake in new location

JosVoigt committed:
Added Doxyfile.in

JosVoigt committed:
updated README and doxygen.cmake to adapt to new structure

JosVoigt committed:
reformatted Parser.h
